RecoTauTag/HLTProducers/src/L1PFJetsMatching.cc

```diff
@@ -0,0 +1,133 @@
+            Output.second.push_back(myJet2);
+            Output.second.push_back(myJetTest);
+             
+		}
+        }
+        
+    }
+    
+            return Output;
+        
+}
+
+L1PFJetsMatching::L1PFJetsMatching(const edm::ParameterSet& iConfig):
+  jetSrc_    ( consumes<PFJetCollection>                     (iConfig.getParameter<InputTag>("JetSrc"      ) ) ),
+  jetTrigger_( consumes<trigger::TriggerFilterObjectWithRefs>(iConfig.getParameter<InputTag>("L1JetTrigger") ) ),
+  pt1_Min   ( iConfig.getParameter<double>("pt1_Min")),
+  pt2_Min   ( iConfig.getParameter<double>("pt2_Min")),
+  mjj_Min   ( iConfig.getParameter<double>("mjj_Min"))
+{  
+  produces<PFJetCollection>("TwoJets");
+  produces<PFJetCollection>("ThreeJets");
+}
+L1PFJetsMatching::~L1PFJetsMatching(){ }
+
+void L1PFJetsMatching::produce(edm::StreamID iSId, edm::Event& iEvent, const edm::EventSetup& iES) const
+{
+    
+  unique_ptr<PFJetCollection> caloL2jets(new PFJetCollection);
+    std::pair<PFJetCollection,PFJetCollection> output;
+    
+  double deltaR    = 1.0;
+  double matchingR = 0.5;
+  
+  // Getting HLT jets to be matched
+  edm::Handle<PFJetCollection > caloJets;
+  iEvent.getByToken( jetSrc_, caloJets );
+        
+  edm::Handle<trigger::TriggerFilterObjectWithRefs> l1TriggeredJets;
+  iEvent.getByToken(jetTrigger_,l1TriggeredJets);
+                
+  //l1t::TauVectorRef jetCandRefVec;
+  l1t::JetVectorRef jetCandRefVec;
+  l1TriggeredJets->getObjects( trigger::TriggerL1Jet,jetCandRefVec);
+
+  math::XYZPoint a(0.,0.,0.);
+        
+ //std::cout<<"PFsize= "<<caloJets->size()<<endl<<" L1size= "<<jetCandRefVec.size()<<std::endl;
+ for(unsigned int iJet = 0; iJet < caloJets->size(); iJet++){
+    for(unsigned int iL1Jet = 0; iL1Jet < jetCandRefVec.size(); iL1Jet++){
+      // Find the relative L2caloJets, to see if it has been reconstructed
+      const PFJet &  myJet = (*caloJets)[iJet];
+    //  if ((iJet<3) && (iL1Jet==0))  std::cout<<myJet.p4().Pt()<<" ";
+      deltaR = ROOT::Math::VectorUtil::DeltaR(myJet.p4().Vect(), (jetCandRefVec[iL1Jet]->p4()).Vect());
+      if(deltaR < matchingR ) {
+        caloL2jets->push_back(myJet);
+        break;
+      }
+    }
+  }  
+   
+    output= categorise(*caloL2jets,pt1_Min,pt2_Min, mjj_Min);
+    unique_ptr<PFJetCollection> output1(new PFJetCollection(output.first));
+    unique_ptr<PFJetCollection> output2(new PFJetCollection(output.second));
+    
+    iEvent.put(std::move(output1),"TwoJets");
+    iEvent.put(std::move(output2),"ThreeJets");
+
+}
+
+void L1PFJetsMatching::fillDescriptions(edm::ConfigurationDescriptions& descriptions) 
+{
+  edm::ParameterSetDescription desc;
+  desc.add<edm::InputTag>("L1JetTrigger", edm::InputTag("hltL1sDoubleIsoTau40er"                     ))->setComment("Name of trigger filter"    );
+  desc.add<edm::InputTag>("JetSrc"      , edm::InputTag("hltSelectedCaloJetsTrackPt1MediumIsolationReg"))->setComment("Input collection of PFJets");
```

Comment:
put meaningful defaults for the input collections, for example the name of you VBF seed filter and the name of the PFJetCollection that you use in the matching